### PR TITLE
Re-use internet_identity.wasm on CI

### PR DESCRIPTION
**Pull request by nmattia**
This creates `cansiter-tests.yml` which reuses the output of the docker
build to run all (backend, selenium) tests.

This reduces the size of the workflows code and saves us some GitHub Actions minutes too. It does add a little overhead (the docker images need to be fully built before the other steps can even start, ~1mn).

This also changes `scripts/docker-build` and the `Dockerfile` to read the `USE_DUMMY_CAPTCHA` environment variable and support the `dummy_captcha` feature; I didn't realize I hadn't actually added it to the docker build.

This will also allow us to add more docker builds more easily, for instance for the upcoming dummy_pow feature and the jailbroken II build.

### CODE REVIEW

**Review by peterpeterparker — APPROVED** [review]

File: .github/workflows/canister-tests.yml [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+    runs-on: ubuntu-latest
+    needs: docker-build-dummy-captcha
+    strategy:
+      matrix:
+        start-flag: [ '', '--emulator' ]
+        device: [ 'desktop', 'mobile' ]
+      # Make sure that one failing test does not cancel all other matrix jobs
+      fail-fast: false
+    env:
+      DFX_VERSION: 0.8.3
```
Comment (lmuntanerDfinity) [inline .github/workflows/canister-tests.yml:158]: why this version? Isn't the latest 0.9.0?

File: .github/workflows/canister-tests.yml [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+      REPLICA_URL: https://ic0.app
```
Comment (lmuntanerDfinity) [inline .github/workflows/canister-tests.yml:159]: Is this running in production?

File: .github/workflows/canister-tests.yml [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+          dfx --version
+          echo node --version
+          node --version
+          echo docker --version
+          docker --version
+
+      # needed so that we can run the test on identity.ic0.app
+      - name: Add hosts to /etc/hosts
+        run: |
+          sudo echo "127.0.0.1 identity.ic0.app" | sudo tee -a /etc/hosts
```
Comment (lmuntanerDfinity) [inline .github/workflows/canister-tests.yml:190]: ah, ok, this redirects the production urls to localhost, I see.

File: .github/workflows/canister-tests.yml [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+
+      - name: Run demo app
+        working-directory: demos/sample-javascript
+        run: |
+          npm ci
+          npm run build
+          npm run develop --no-live-reload --no-devtool &
+          sleep 10 # crude
+
+      - name: install selenium webdrivers
```
Comment (lmuntanerDfinity) [inline .github/workflows/canister-tests.yml:227]: Is this running from the same working directory as the previous step?

File: .github/workflows/canister-tests.yml [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+      - run: |
+          echo the following screenshots were recovered
+          find ./screenshots -name '*.png' -maxdepth 1 -type f -print0 | sort -z | xargs -r0 shasum -a 256
+
+      - name: Commit screenshots
+        uses: EndBug/add-and-commit@v7.4.0
+        with:
+          add: screenshots
+          author_name: Screenshot Committer
+          author_email: "<nobody@example.com>"
```
Comment (lmuntanerDfinity) [inline .github/workflows/canister-tests.yml:288]: 😢  nobody?

File: .github/workflows/canister-tests.yml (hunk 4 above)
Comment (nmattia) [reply .github/workflows/canister-tests.yml:227]: nope it's not! each step runs in the (repo) root directory by default

File: .github/workflows/canister-tests.yml (hunk 2 above)
Comment (nmattia) [reply .github/workflows/canister-tests.yml:159]: FTR: https://github.com/dfinity/internet-identity/pull/532#discussion_r802305081

File: .github/workflows/canister-tests.yml (hunk 1 above)
Comment (nmattia) [reply .github/workflows/canister-tests.yml:158]: because we've been using this so far, will update in another PR!

**Pull request merged**